homeassistant/components/nextcloud/sensor.py

```diff
@@ -0,0 +1,172 @@
+    "system_cpuload": ["nextcloud", "system", "cpuload"],
+    "system_mem_total": ["nextcloud", "system", "mem_total"],
+    "system_mem_free": ["nextcloud", "system", "mem_free"],
+    "system_swap_total": ["nextcloud", "system", "swap_total"],
+    "system_swap_free": ["nextcloud", "system", "swap_free"],
+    "system_apps_num_installed": ["nextcloud", "system", "apps", "num_installed"],
+    "system_apps_num_updates_available": [
+        "nextcloud",
+        "system",
+        "apps",
+        "num_updates_available",
+    ],
+    "system_apps_app_updates": ["nextcloud", "system", "apps", "app_updates"],
+    "storage_num_users": ["nextcloud", "storage", "num_users"],
+    "storage_num_files": ["nextcloud", "storage", "num_files"],
+    "storage_num_storages": ["nextcloud", "storage", "num_storages"],
+    "storage_num_storages_local": ["nextcloud", "storage", "num_storages_local"],
+    "storage_num_storages_home": ["nextcloud", "storage", "num_storages_home"],
+    "storage_num_storages_other": ["nextcloud", "storage", "num_storages_other"],
+    "storage_num_shares": ["nextcloud", "shares", "num_shares"],
+    "storage_num_shares_user": ["nextcloud", "shares", "num_shares_user"],
+    "storage_num_shares_groups": ["nextcloud", "shares", "num_shares_groups"],
+    "storage_num_shares_link": ["nextcloud", "shares", "num_shares_link"],
+    "storage_num_shares_mail": ["nextcloud", "shares", "num_shares_mail"],
+    "storage_num_shares_room": ["nextcloud", "shares", "num_shares_room"],
+    "storage_num_shares_link_no_password": [
+        "nextcloud",
+        "shares",
+        "num_shares_link_no_password",
+    ],
+    "storage_num_fed_shares_sent": ["nextcloud", "shares", "num_fed_shares_sent"],
+    "storage_num_fed_shares_received": [
+        "nextcloud",
+        "shares",
+        "num_fed_shares_received",
+    ],
+    "server_webserver": ["server", "webserver"],
+    "server_php_version": ["server", "php", "version"],
+    "server_php_memory_limit": ["server", "php", "memory_limit"],
+    "server_php_max_execution_time": ["server", "php", "max_execution_time"],
+    "server_php_max_file_size": ["server", "php", "upload_max_filesize"],
+    "server_database_type": ["server", "database", "type"],
+    "server_database_version": ["server", "database", "version"],
+    "server_database_size": ["server", "database", "size"],
+    "active_users_last_5_minutes": ["activeUsers", "last5minutes"],
+    "active_users_last_1_hour": ["activeUsers", "last1hour"],
+    "active_users_last_24_hours": ["activeUsers", "last24hours"],
+}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None) -> None:
+    """Set up the Nextcloud sensor."""
+    # Fetch Nextcloud Monitor api data
+    ncm = NextcloudMonitor(
+        config[CONF_URL], config[CONF_USERNAME], config[CONF_PASSWORD]
+    )
+    if not ncm.data:
+        _LOGGER.error("No data returned from Nextcloud API. Check your user/password.")
```

Comment:
Please end your logs without a period.
```suggestion
        _LOGGER.error("No data returned from Nextcloud API, please check your credentials")
```

Reply:
This error check can actually be removed since there is a similar check upstream in the pypi package.